apps/src/maze/maze.js

```diff
@@ -372,6 +373,9 @@ module.exports = class Maze {
             this.onExecutionFinish_();
             if (this.executionInfo.terminationValue() === true) {
               successes.push(i);
+            } else if (this.executionInfo.terminationValue() === Infinity) {
```

Comment:
I'd really like to see a comment here explaining both what this termination value indicates and why we're treating it differently.

I'd also be interested to see some reasoning about what we want to do in the situation in which the user encounters an infinite loop on only _some_ of the possible grids. Specifically with regards to deciding which failure case to show them